src/plugins/minimal_scene/MinimalSceneRhiOpenGL.cc

```diff
@@ -0,0 +1,282 @@
+void RenderThreadRhiOpenGL::RenderNext(RenderSync *_renderSync)
+{
+  this->dataPtr->context->makeCurrent(this->dataPtr->surface);
+
+  if (!this->dataPtr->renderer->initialized)
+  {
+    this->dataPtr->renderer->Initialize();
+  }
+
+  if (!this->dataPtr->renderer->initialized)
+  {
+    ignerr << "Unable to initialize renderer" << std::endl;
+    return;
+  }
+
+  // Call the renderer
+  this->dataPtr->renderer->Render(_renderSync);
+
+  // Get reference to the rendered texture
+  this->dataPtr->texturePtr = nullptr;
+  this->dataPtr->renderer->TextureId(&this->dataPtr->texturePtr);
+
+  this->dataPtr->context->doneCurrent();
+}
+
+/////////////////////////////////////////////////
+void* RenderThreadRhiOpenGL::TexturePtr() const
+{
+  return this->dataPtr->texturePtr;
+}
+
+/////////////////////////////////////////////////
+QSize RenderThreadRhiOpenGL::TextureSize() const
+{
+  return this->dataPtr->renderer->textureSize;
+}
+
+/////////////////////////////////////////////////
+void RenderThreadRhiOpenGL::ShutDown()
+{
+  this->dataPtr->renderer->Destroy();
+
+  this->dataPtr->texturePtr = nullptr;
+
+  this->dataPtr->context->doneCurrent();
+  delete this->dataPtr->context;
+  this->dataPtr->context = nullptr;
+
+  // Schedule this to be deleted only after we're done cleaning up
+  this->dataPtr->surface->deleteLater();
+}
+
+/////////////////////////////////////////////////
+/////////////////////////////////////////////////
+TextureNodeRhiOpenGL::~TextureNodeRhiOpenGL()
+{
+  delete this->dataPtr->texture;
+  this->dataPtr->texture = nullptr;
+}
+
+/////////////////////////////////////////////////
+TextureNodeRhiOpenGL::TextureNodeRhiOpenGL(QQuickWindow *_window)
+    : dataPtr(std::make_unique<TextureNodeRhiOpenGLPrivate>())
+{
+  this->dataPtr->window = _window;
+
+  // Our texture node must have a texture, so use the default 0 texture.
+#if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
+# ifndef _WIN32
+#   pragma GCC diagnostic push
+#   pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+# endif
+  this->dataPtr->texture = this->dataPtr->window->createTextureFromId(
+      this->dataPtr->textureId,
+      QSize(1, 1),
+      QQuickWindow::TextureIsOpaque);
+# ifndef _WIN32
+#   pragma GCC diagnostic pop
+# endif
+#else
+  this->dataPtr->texture =
+      this->dataPtr->window->createTextureFromNativeObject(
+          QQuickWindow::NativeObjectTexture,
+          static_cast<void*>(&this->dataPtr->textureId),
+          0,
+          QSize(1, 1));
+#endif
+}
+
+/////////////////////////////////////////////////
+QSGTexture *TextureNodeRhiOpenGL::Texture() const
+{
+  return this->dataPtr->texture;
+}
+
+/////////////////////////////////////////////////
+bool TextureNodeRhiOpenGL::HasNewTexture() const
+{
+  return (this->dataPtr->newTextureId != 0);
+}
+
+/////////////////////////////////////////////////
+void TextureNodeRhiOpenGL::NewTexture(
+    void* _texturePtr /*[in]*/, const QSize &_size)
```

Comment:
Remove `/*[in]*/`?

Reply:
Updated in ca52977561c63bb615e01b25fa62b350645dfda6